bench/state_transition/process_epoch.zig

```diff
@@ -0,0 +1,491 @@
+// pending_consolidations 41       2.185s         53.298ms ± 12.57ms     (45.366ms ... 122.706ms)     54.65ms    122.706ms  122.706ms
+// effective_balance_upda 31       1.586s         51.183ms ± 4.754ms     (47.418ms ... 71.893ms)      53.738ms   71.893ms   71.893ms
+// slashings_reset        38       1.875s         49.351ms ± 5.82ms      (44.875ms ... 73.459ms)      52.233ms   73.459ms   73.459ms
+// randao_mixes_reset     27       1.342s         49.726ms ± 4.887ms     (44.524ms ... 59.195ms)      52.718ms   59.195ms   59.195ms
+// historical_summaries   41       1.926s         46.978ms ± 2.409ms     (45.1ms ... 55.991ms)        47.013ms   55.991ms   55.991ms
+// participation_flags    63       1.865s         29.617ms ± 1.139ms     (27.794ms ... 33.642ms)      30.019ms   33.642ms   33.642ms
+// sync_committee_updates 61       1.877s         30.784ms ± 2.689ms     (27.568ms ... 40.1ms)        31.492ms   40.1ms     40.1ms
+// proposer_lookahead     16       1.595s         99.687ms ± 5.406ms     (92.066ms ... 114.437ms)     104.102ms  114.437ms  114.437ms
+//
+// Benchmark for epoch processing (Fulu)
+// https://github.com/ethereum/consensus-specs/blob/master/specs/phase0/beacon-chain.md#epoch-processing
+//
+// Benchmarks all process_epoch operations per spec:
+// - process_justification_and_finalization
+// - process_inactivity_updates
+// - process_rewards_and_penalties
+// - process_registry_updates
+// - process_slashings
+// - process_eth1_data_reset
+// - process_pending_deposits
+// - process_pending_consolidations
+// - process_effective_balance_updates
+// - process_slashings_reset
+// - process_randao_mixes_reset
+// - process_historical_summaries_update
+// - process_participation_flag_updates
+// - process_sync_committee_updates
+// - process_proposer_lookahead
+
+const CachedBeaconStateAllForks = state_transition.CachedBeaconStateAllForks;
+const BeaconStateAllForks = state_transition.BeaconStateAllForks;
+const EpochTransitionCache = state_transition.EpochTransitionCache;
+const ValidatorIndex = types.primitive.ValidatorIndex.Type;
+const PubkeyIndexMap = state_transition.PubkeyIndexMap(ValidatorIndex);
+const BeaconState = types.fulu.BeaconState;
+
+const ProcessJustificationAndFinalizationBench = struct {
+    allocator: std.mem.Allocator,
+    cached_state: *CachedBeaconStateAllForks,
+
+    pub fn run(self: ProcessJustificationAndFinalizationBench, allocator: std.mem.Allocator) void {
+        _ = allocator;
+        const cloned = self.cached_state.clone(self.allocator) catch return;
+        defer {
+            cloned.deinit();
+            self.allocator.destroy(cloned);
+        }
+        var cache = EpochTransitionCache.init(self.allocator, cloned) catch return;
+        defer {
+            cache.deinit();
+            self.allocator.destroy(cache);
+        }
+        state_transition.processJustificationAndFinalization(cloned, cache) catch return;
+    }
+};
```

Comment:
```suggestion
const ProcessJustificationAndFinalizationBench = struct {
    cached_state: *CachedBeaconStateAllForks,

    pub fn run(self: ProcessJustificationAndFinalizationBench, allocator: std.mem.Allocator) void {
        const cloned = self.cached_state.clone(allocator) catch return;
        defer {
            cloned.deinit();
            allocator.destroy(cloned);
        }
        var cache = EpochTransitionCache.init(allocator, cloned) catch return;
        defer {
            cache.deinit();
            allocator.destroy(cache);
        }
        state_transition.processJustificationAndFinalization(cloned, cache) catch return;
    }
};
```

We can probably do away with the `allocator` in the struct and just use the passed in `allocator`, same goes for the other benchmark structs.